remoteappmanager/handlers/base_handler.py

```diff
@@ -40,3 +41,20 @@ def render(self, template_name, **kwargs):
 
         args.update(kwargs)
         super(BaseHandler, self).render(template_name, **args)
+
+    def write_error(self, status_code, **kwargs):
+        """Render error page for uncaught errors"""
+        status_message = responses.get(status_code, 'Unknown HTTP Error')
+        message = ""
+
+        # If this error was caused by an uncaught exception
+        # log exception message and reference number as well
+        exc_info = kwargs.get('exc_info')
+        if exc_info:
+            exception = exc_info[1]
+            ref = self.log.issue(status_message, exception)
+            reason = getattr(exception, 'reason', '')
+            message = '{} Ref.: {}'.format(reason, ref)
```

Comment:
By default, `web.HTTPError.reason` is empty.  If it is empty, the `str(HTTPError)` representation is deduced in by the status code (e.g. 404 -> Not Found) .  If someone raises an HTTPError with a `reason` attribute, that content would be displayed to the user *even if the `write_error`* was not overload.  So it is entirely the web developer's choice what to put in `reason`.
![screenshot from 2016-07-27 17 15 19](https://cloud.githubusercontent.com/assets/3673984/17183055/cc90b020-541d-11e6-904f-f148257ebc21.png)